lib/Emitter.php

```diff
@@ -25,32 +26,32 @@
             /**
              * @var callable
              */
-            private $resolve;
+            private $complete;
 
             /**
              * @var callable
              */
             private $fail;
 
             public function __construct() {
-                $this->stream = new Internal\PrivateStream(
-                    function (callable $emit, callable $resolve, callable $fail) {
+                $this->iterator = new Internal\PrivateIterator(
+                    function (callable $emit, callable $complete, callable $fail) {
                         $this->emit = $emit;
-                        $this->resolve = $resolve;
+                        $this->complete = $complete;
                         $this->fail = $fail;
                     }
                 );
             }
 
             /**
-             * @return \Amp\Stream
+             * @return \Amp\Iterator
              */
-            public function stream(): Stream {
-                return $this->stream;
+            public function getIterator(): Iterator {
```

Comment:
I would also prefer `Emitter::iterate()` here, partly for symmetry but also because @bwoebi makes a good point with `IteratorAggregate` returning a new instance, so too would be the assumption here with `getIterator()`.

I think docs will just need to be clear here regardless of which name is chosen, so in that case symmetry wins out IMO.